onegov/election_day/formats/election/wabstic_proporz.py

```diff
@@ -69,34 +69,99 @@ def line_is_relevant(line, number, district=None):
         return line.sortgeschaeft == number
 
 
-def get_entity_id(line, entities):
-    entity_id = int(line.bfsnrgemeinde or 0)
-    return 0 if entity_id in EXPATS else entity_id
-
-
-def get_candidate_id(line):
-    return str(int(line.knr))
+def get_number_of_mandates(line):
+    col = 'sitze'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.sitze or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_ausmittlungsstand(line):
+    col = 'ausmittlungsstand'
+    try:
+        t = int(line.ausmittlungsstand or 0)
+        assert 0 <= t <= 3
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+    except AssertionError:
+        raise AssertionError(
+            _('Value ${col} is not between 0 and 3', mapping={'col': col}))
+
+
+def get_stimmberechtigte(line):
+    col = 'stimmberechtigte'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.stimmberechtigte or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_stimmentotal(line):
+    col = 'stimmentotal'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.stimmentotal)
+    except ValueError:
+        raise ValueError(_('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_entity_id(line, expats):
+    if not hasattr(line, 'bfsnrgemeinde'):
+        raise ValueError(
+            _('Missing column: bfsnrgemeinde'))
+    try:
+        entity_id = int(line.bfsnrgemeinde or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: bfsnrgemeinde'
+              ' Can not extract entity_id.'))
+    return 0 if entity_id in expats else entity_id
+
+
+def get_list_id_from_knr(line):
+    """
+    Takes a line from csv file with a candidate number (knr) in it and
+    returns the derived listnr for this candidate. Will also handle the new
+    WabstiC Standard 2018.
+    """
+    col = 'knr'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col', col}))
+    if '.' in line.knr:
+        return line.knr.split('.')[0]
+    # replaces int(int(line.knr) / 100))
```

Comment:
Is this the previous function in place? If so, you can just remove it. Only leave the current code when changing functions. It is hard enough to understand old code when you look at it in a year, without having to look at even older code.

If you want to look at even older code, git log / git blame is better.

Reply:
Remove the old code hint, old code is not used anywhere anymore.